### PR TITLE
SAIC-803 Glance images members as required tenants

### PR DESCRIPTION
**Pull request by aguzikova**
Tenants required to be migrated before migration of glance images
    are all the members of glance images, not only owners.
    
    If a tenant-member is not migrated then a membership update fails.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:398: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:483: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:493: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/image/glance_image.py [hunk 1]
```diff
@@ -373,24 +386,29 @@ def make_image_info(self, glance_image, info):
...
                 dst_image.owner, default=self.cloud.cloud_config.cloud.tenant)
             image_key = (dst_image.name, tenant_name, dst_image.checksum,
                          dst_image.is_public)
             dst_images[image_key] = dst_image
+        return dst_images
+
+    def deploy(self, *args):
+        LOG.info("Glance images deployment started...")
+        if len(args) != 1:
+            raise TypeError("deploy() takes exactly 1 argument.")
```
Comment (roman-verchikov) [inline cloudferrylib/os/image/glance_image.py]: Please revert to old `deploy()` signature, these types of checks must be done by interpreter, not by business logic.

**Comment by roman-verchikov** [comment]
Looks ok other than one inline. Please also update commit message and pull request detailed description to have more information on the patch.

File: cloudferrylib/os/image/glance_image.py (hunk 1 above)
Comment (aguzikova) [inline cloudferrylib/os/image/glance_image.py]: This fixes a pylint issue: signature differs from overridden method.

Comment (roman-verchikov) [inline cloudferrylib/os/image/glance_image.py]: In fact it doesn't, instead it only hides the issue. The resolution for that would be to refactor all inherited `deploy()` methods to follow the same signature. Again, type checks must be done by interpreter, not the business logic.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/base/resource.py [hunk 2]
```diff
@@ -24,15 +24,16 @@ class Resource(object):
...
         retry = cfg.migrate.retry
         time_wait = cfg.migrate.time_wait
         return proxy_client.Proxy(client, retry, time_wait)
 
-    def read_info(self, opts=None):
+    def read_info(self, **search_opts):
         pass
 
-    def deploy(self, *args):
+    def deploy(self, info):
```
Comment (roman-verchikov) [inline cloudferrylib/base/resource.py]: Following implementations will not override this method correctly:

 - cloudferrylib.os.compute.nova_compute.NovaCompute#deploy
 - cloudferrylib.os.object_storage.swift_storage.SwiftStorage#deploy

Let's leave this pylint issue for more global refactoring

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:393: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
+1

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**